catalogue/webapp/services/catalogue/images.js

```diff
@@ -6,30 +6,39 @@ import type {
 } from '@weco/common/model/catalogue';
 import { type CatalogueImagesApiProps } from '@weco/common/services/catalogue/api';
 import { serialiseUrl } from '@weco/common/services/catalogue/routes';
-import { type Environment, rootUris } from './common';
+import { type Toggles, rootUris, globalApiOptions } from './common';
 
 type GetImagesProps = {|
   params: CatalogueImagesApiProps,
   pageSize?: number,
-  ...Environment,
+  toggles: Toggles,
 |};
 
 type GetImageProps = {|
   id: string,
-  ...Environment,
+  toggles: Toggles,
 |};
 
 export async function getImages({
   params,
-  env = 'prod',
+  toggles,
   pageSize = 25,
 }: GetImagesProps): Promise<CatalogueResultsList<Image> | CatalogueApiError> {
-  const filterQueryString = Object.keys(serialiseUrl(params)).map(key => {
-    const val = params[key];
-    return `${key}=${encodeURIComponent(val)}`;
-  });
+  const apiOptions = globalApiOptions(toggles);
+  const extendedParams = {
+    ...params,
+    _index: apiOptions.indexOverrideSuffix
+      ? `images-${apiOptions.indexOverrideSuffix}`
+      : undefined,
```

Comment:
We have used `null` everywhere else, and this is going to start fragmenting this.

Could we either say why we're using undefined as a different pattern, or, with the good logic behind it, get it across the code base?

Reply:
Sorry it's just habit rather than a conscious choice. One thing I was going to say the other day is that `null` will cause problems if/when we might want to migrate to Typescript - eg a `?string` in TS means `string | undefined` and you would need to manually write `string | null` to allow nulls. Following from this the official TS style guide advises against any usage of `null`.

Reply:
Completely agree with doing it that way, `null` is crap. Do you think it's fine to start doing it this way piecemeal?

https://github.com/wellcomecollection/wellcomecollection.org/issues/5362

Reply:
I think we probably want to do it all at once. We also have a mix of `{ foo?: bar }` and `{ foo: ?bar }` which can be really confusing/irritating when you need to explicitly set stuff as nullish - maybe standardise that (to the former?) in the same ticket?